zerocopy-derive/tests/ui-nightly/late_compile_pass.rs

```diff
@@ -72,3 +73,17 @@ struct Unaligned2 {
 struct Unaligned3 {
     aligned: AU16,
 }
+
+//
+// KnownLayout errors
+//
+
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          Y |        Y |              N |        Y |      KL13 |
+#[derive(KnownLayout)]
+#[repr(C)]
+struct KL13<T>(u8, T);
+
+fn test_kl13<T>(t: T) -> impl KnownLayout {
+    KL13(0u8, t)
+}
```

Comment:
Why does this type only generate errors on the nightly toolchain?

Reply:
More compiler pass weirdness. I've move the test to `mid_compile_pass.rs`, and it seems to produce an error on all toolchains. It's unfortunate how fragile this all is. We really should revisit the `ui-test` crate.

Reply:
Agreed.